### ISSUE TITLE
DayToSecond generates nanosecond precision for DAY_MICROSECOND interval

### ISSUE THREAD

**Issue by danadam** [jOOQ/jOOQ#6820]
### Expected behavior and actual behavior:
The following code:
```java
DayToSecond exp = DayToSecond.valueOf(1300);
...
...Tables.SOME_TABLE.SOME_DATETIME_FIELD.add(exp)...
```
generates the following sql:
```sql
date_add(`some_table`.`some_datetime_field`, interval '+0 00:00:01.300000000' day_microsecond)
```
Mysql parses that as 1 second and 300'000'000 microseconds and gives 301 second interval. If the sql had 3 zeros less:
```sql
date_add(`some_table`.`some_datetime_field`, interval '+0 00:00:01.300000' day_microsecond)
```
it would be parsed correctly as 1.3 seconds.

### Versions:

- jOOQ: 3.10.1
- Java: 1.8.0_151
- Database (include vendor): mysql 5.7, mariadb 10.1
- JDBC Driver (include name if inofficial driver): mysql-connector-java-5.1.44-bin.jar, mariadb-java-client-2.2.0.jar

**Comment by lukaseder** [comment]
Oh wow, thanks for reporting this, quite nasty. Have you reported this to MySQL as well? There's no doubt about the meaning of fractional seconds. This must not produce the value it did

**Comment by danadam** [comment]
There you have it: [mysql](https://bugs.mysql.com/bug.php?id=88573), [mariadb](https://jira.mariadb.org/browse/MDEV-14452)

**Comment by lukaseder** [comment]
Excellent, thank you very much.

**Comment by lukaseder** [comment]
Yeah, our integration tests obviously didn't check for fractional seconds - what a shame. This is now fixed on master and will be merged to 3.10.2 (#6830) and 3.9.7 (#6831)

**Comment by lukaseder** [comment]
Thanks again for reporting!
